portal/views/auth.py

```diff
@@ -276,6 +276,15 @@ def picture_url(result):
         from werkzeug.datastructures import ImmutableMultiDict
         request.args = ImmutableMultiDict()
 
+    prv = 'FB' if (provider_name == 'facebook') else provider_name.upper()
+
+    if (not current_app.config.get('{}_CONSUMER_KEY'.format(prv)) or
```

Comment:
the second arg defines a default - which is itself `None` if not provided.  this code looks to be doing what we need, to me.